internal/services/compute/capacity_reservation_resource.go

```diff
@@ -0,0 +1,207 @@
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := parse.NewCapacityReservationID(subscriptionId, d.Get("resource_group_name").(string), d.Get("capacity_name").(string), d.Get("name").(string))
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for presence of %s: %+v", id, err)
+			}
+		}
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_capacity_reservation", id.ID())
+		}
+	}
+
+	parameters := compute.CapacityReservation{
+		Location: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
+		Sku:      expandCapacityReservationSku(d.Get("sku").([]interface{})),
+		Tags:     tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	if v, ok := d.GetOk("zones"); ok {
+		parameters.Zones = utils.ExpandStringSlice(v.([]interface{}))
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name, parameters); err != nil {
+		return fmt.Errorf("creating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceCapacityReservationRead(d, meta)
+}
+
+func resourceCapacityReservationRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CapacityReservationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO]  Capacity Reservation %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("reading %q: %+v", id, err)
+	}
+
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("capacity_name", id.CapacityReservationGroupName)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+
+	d.Set("zones", resp.Zones)
+
+	if err := d.Set("sku", flattenCapacityReservationSku(resp.Sku)); err != nil {
+		return fmt.Errorf("setting `sku`: %+v", err)
+	}
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceCapacityReservationDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.CapacityReservationsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.CapacityReservationID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if _, err := client.Delete(ctx, id.ResourceGroup, id.CapacityReservationGroupName, id.Name); err != nil {
+		return fmt.Errorf("deleting %q: %+v", id, err)
+	}
+
+	return nil
+}
+
+func expandCapacityReservationSku(input []interface{}) *compute.Sku {
```

Comment:
add check on input, it might be empty, example: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/springcloud/spring_cloud_configuration_service_resource.go#L233